### DISCUSSION TITLE
recommended way to pass audio from other process to vst output

### DISCUSSION THREAD

**Discussion by SeleDreams** [DISTRHO/DPF#486] [Q&A]
Hi, I've been with dpf programming a plugin for OpenUTAU allowing to use openutau in a daw by using TCP to transmit the time of the daw, the play status etc
here's the current status https://youtu.be/ybWCgldQiF4

however right now the audio is directly played by openutau, i'm looking for the most efficient way to send the audio directly to the vst.
Do you have any ideas about the best way to go about it ?
the vst is C++ but openutau is C# under .net core

**Comment by falkTX** [comment]
So the audio comes from a separate process?
Shared memory + semaphores is likely fastest.

**Comment by SeleDreams** [comment]
> So the audio comes from a separate process? Shared memory + semaphores is likely fastest.

can it be done if the other process is in c# tho ? i think it might be trickier

**Comment by falkTX** [comment]
I have no experience with that, so cant really tell.
